frontend/src/metabase/home/homepage/components/Homepage/Homepage.jsx

```diff
@@ -0,0 +1,71 @@
+import React, { Fragment } from "react";
+import PropTypes from "prop-types";
+import GreetingSection from "../GreetingSection";
+import CollectionSection from "../CollectionSection";
+import DatabaseSection from "../DatabaseSection";
+import StartSection from "../StartSection";
+import XraySection from "../XraySection";
+import { LandingRoot } from "./Homepage.styled";
+
+const propTypes = {
+  user: PropTypes.object.isRequired,
+  databases: PropTypes.array,
+  collections: PropTypes.array,
+  dashboards: PropTypes.array,
+  databaseCandidates: PropTypes.array,
+  showData: PropTypes.bool,
+  showXrays: PropTypes.bool,
+  showPinMessage: PropTypes.bool,
+  onHideData: PropTypes.func,
+  onHideXrays: PropTypes.func,
+  onHidePinMessage: PropTypes.func,
+};
+
+const Homepage = ({
+  user,
+  databases,
+  collections,
+  dashboards,
+  databaseCandidates,
+  showData,
+  showXrays,
+  showPinMessage,
+  onHideData,
+  onHideXrays,
+  onHidePinMessage,
+}) => {
+  return (
+    <LandingRoot>
+      <GreetingSection user={user} />
+      {databases && collections && dashboards && (
```

Comment:
So now everything waits to be shown until data for all sections is loaded. Is there an opportunity to split that?

Reply:
We need this data to understand what content to show in each section, so unfortunately there is not much we can do here. The only thing that is not required is xrays content, and it's lazy loaded.